scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/MockBigQuery.scala

```diff
@@ -70,23 +70,29 @@ class MockBigQuery private (private val bq: BigQuery) {
    */
   def mockWildcardTable(prefix: String, suffix: String): MockTable = {
     val original = BigQueryHelpers.parseTableSpec(prefix + suffix)
+    require(
+      !mapping.contains(original),
+      s"Table ${BigQueryHelpers.toTableSpec(original)} already registered for mocking"
+    )
+
+    // fake table reference, only used in the mapping for query replacement
+    val t = bq.tables.table(original)
     val wildcard = new TableReference()
       .setProjectId(original.getProjectId)
       .setDatasetId(original.getDatasetId)
       .setTableId(original.getTableId.dropRight(suffix.length) + "*")
-
-    require(
-      !mapping.contains(wildcard),
-      s"Table ${BigQueryHelpers.toTableSpec(wildcard)} already registered for mocking"
+    val tempWildcard = mapping.getOrElseUpdate(
+      wildcard, {
+        // fake table reference, only used in the mapping for query replacement
+        val w = bq.tables.temporaryTableReference(t.getLocation)
+        w.setTableId(w.getTableId + "_*")
+        w
+      }
     )
 
-    val t = bq.tables.table(original)
-    val temp = bq.tables.createTemporary(t.getLocation, Some(suffix)).getTableReference
-    val tmpWildcard = new TableReference()
-      .setProjectId(temp.getProjectId)
-      .setDatasetId(temp.getDatasetId)
-      .setTableId(temp.getTableId.dropRight(suffix.length) + "*")
-    mapping += (wildcard -> tmpWildcard)
+    val ref = tempWildcard.clone().setTableId(tempWildcard.getTableId.dropRight(1) + suffix)
```

Comment:
maybe I'm missing something, how is `ref` different from `original`?

Reply:
ref is actually pointing to the test table in the staging dataset:
- prefix: `bigquery-public-data.noaa_gsod.gsod20`
- suffix: `20`
- original: `bigquery-public-data.noaa_gsod.gsod2020`
- wildcard: `bigquery-public-data.noaa_gsod.gsod20*`
- tempWildcard: `my-test-project.scio_bigquery_staging_us_east_1.scio_query_20210422164412_*`
- ref:  `my-test-project.scio_bigquery_staging_us_east_1.scio_query_20210422164412_20`

Reply:
could also be named `temp` since we do not have to get the reference after creation

Reply:
ah gotcha! Maybe rename to `tempRef`? 🎨  Otherwise looks good!